src/main/java/com/minecolonies/coremod/commands/CommandEntryPointNew.java

```diff
@@ -267,15 +278,16 @@ public void execute(@NotNull final MinecraftServer server, @NotNull final IComma
         }
     }
 
-    protected void createInstanceAndExecute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final ActionMenu actionMenu,
+    protected void createInstanceAndExecute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final ActionMenuState actionMenuState,
             @NotNull final Class<? extends IActionCommand> clazz) throws InstantiationException, IllegalAccessException, CommandException
     {
         final IActionCommand actionCommand = clazz.newInstance();
-        actionCommand.execute(server, sender, actionMenu);
+        actionCommand.execute(server, sender, actionMenuState);
     }
 
     private static void throwCommandUsageExceptionIfRequiredArgumentsAreNotProvided(@NotNull final TreeNode<IMenu> executionTreeNode, @NotNull final ActionMenu actionMenu,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 34 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)